rundeckapp/src/test/groovy/rundeck/controllers/PluginControllerSpec.groovy

```diff
@@ -265,6 +265,48 @@ class PluginControllerSpec extends Specification implements ControllerUnitTest<P
             'WebhookEvent'   | 2
     }
 
+    def "plugin list returns isHighlighted and highlightedOrder properties"() {
+        given:
+        controller.pluginService = Mock(PluginService)
+        controller.pluginApiService = Mock(PluginApiService)
+        controller.uiPluginService = Mock(UiPluginService)
+        params.service = null
+
+        def plugins = [
+                [
+                        service  : 'WebhookEvent',
+                        providers: [
+                                [isHighlighted   : true,
+                                 highlightedOrder: 123
+                                ],
+                                [:]],
+                ],
+                [
+                        service  : 'Foo',
+                        providers: [
+                                [isHighlighted   : Boolean.FALSE,
+                                 highlightedOrder: Integer.valueOf(321)
```

Comment:
Tiny nit: I'd suggest to remove the highlightedOrder here to represent the plugins that do not have that set (so that the default value would be returned)

Reply:
IMO, it's a sub-standard practice to deal with data inconsistencies in the presentation layer (API response layer). Instead, this layer should reflect the state of the model -- a single source of truth. Model invariant violations should be checked and enforced in the model.

Reply:
But that's exactly why I suggested to leave a plugin without setting an order value, to reflect the state of the model as not all plugins have an order value defined (if they are not highlighted, the code would return the default value). Take a look on the AzurePluginGroup for example.

Reply:
What problem are we trying to solve here? This is the mock data we are looking at here?

Reply:
Context: We have more than 100 plugins, and only 4-5 are actually highlighted and have a value for order other than the default value.

The "problem": I'm suggesting to change the mock data to match the scenarios that the API will face, that is: 
- a plugin is highlighted, and has an order value;
- a plugin isn't highlighted, and therefore there's no order value defined, so API will be returning the default value for order;

Reply:
Marking as resolved as I missed the `[:]` in the first array

Reply:
Both scenarios are tested:

> a plugin is highlighted, and has an order value;
https://github.com/rundeck/rundeck/pull/9464/files#diff-8b683baffd417ea324face5542fe52d1e39d676918e24d2e703660b6e96cf8ecR279

> a plugin isn't highlighted, and therefore there's no order value defined, so API will be returning the default value for order
https://github.com/rundeck/rundeck/pull/9464/files#diff-8b683baffd417ea324face5542fe52d1e39d676918e24d2e703660b6e96cf8ecR282